src/cleanfigure.m

```diff
@@ -251,78 +250,161 @@ function pruneOutsideBox(meta, handle)
+      X1_first  = data(id_first,    :);
+      X2_first  = data(id_first+1,  :);
+      X1_second = data(id_second,   :);
+      X2_second = data(id_second-1, :);
+
+      % Move the points closer to the large box along the segment
+      newData_first = moveToBox(X1_first,  X2_first,  largeXLim, largeYLim);
+      newData_second= moveToBox(X1_second, X2_second, largeXLim, largeYLim);
+      
+      % Respect logarithmic scaling for the new points
+      isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
+      if isXlog
+          newData_first (:, 1) = 10.^newData_first (:, 1);
+          newData_second(:, 1) = 10.^newData_second(:, 1);
+      end
+      isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
+      if isYlog
+          newData_first (:, 2) = 10.^newData_first (:, 2);
+          newData_second(:, 2) = 10.^newData_second(:, 2);
+      end
+
+      % If newData_* is infinite, the segment was not visible. However, as we
+      % move the point closer, it would become visible. So insert a NaN.
+      isInfinite_first  = any(~isfinite(newData_first),  2);
+      isInfinite_second = any(~isfinite(newData_second), 2);
+
+      newData_first(isInfinite_first,  :) = NaN(sum(isInfinite_first),  2);
+      newData_second(isInfinite_second,:) = NaN(sum(isInfinite_second), 2);
+
+      % If a point is part of two segments, that cross the border, we need to
+      % insert a NaN to prevent an additional line segment
+      [trash, trash, id_conflict] = intersect(id_first (~isInfinite_first), ...
+                                              id_second(~isInfinite_second));
+
+      % Cut the data into length(id_replace)+1 segments.
+      % Calculate the length of the segments
+      length_segments = [id_replace(1);
+                         diff(id_replace);
+                         size(data, 1)-id_replace(end)];
+
+      % Create an empty cell array for inserting NaNs and fill it at the
+      % conflict sites
+      dataInsert_NaN              = cell(length(length_segments),1);
+      dataInsert_NaN(id_conflict) = mat2cell(NaN(length(id_conflict), 2),...
+                                             ones(size(id_conflict)), 2);
+
+      % Create a cell array for the moved points
+      dataInsert_first  = mat2cell(newData_first,  ones(size(id_first)),  2);
+      dataInsert_second = mat2cell(newData_second, ones(size(id_second)), 2);
+
+      % Add an empty cell at the end of the last segment
+      dataInsert_first  = [dataInsert_first;  cell(1)];
+      dataInsert_second = [dataInsert_second; cell(1)];
+
+      % If the first or the last point would have been replaced add an empty
+      % cell at the beginning/end
+      if(id_replace(end) == size(data, 1))
+        dataInsert_first  = [dataInsert_first; cell(1)];
+      end
+      if(id_replace(1) == 1)
+        dataInsert_second = [cell(1); dataInsert_second];
+      end
+
+      % Put the cells together, right points first, then the possible NaN
+      % and then the left points      
```

Comment:
Hm in the old code it was easier as I directly transformed that into a simple matrix. Here it is more difficult due to the new functions.

I could just merge the arrays and pass them to dataInsert, however it would be more difficult to get the number of inserted data points